src/MassTransit/Transports/Outbox/OutboxSendTransport.cs

```diff
@@ -0,0 +1,113 @@
+﻿using GreenPipes;
+using MassTransit.Context;
+using MassTransit.Logging;
+using MassTransit.Serialization;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MassTransit.Transports.Outbox
+{
+    public class OutboxSendTransport : ISendTransport
+    {
+        private readonly IOutboxSendTransportContext _context;
+
+        public OutboxSendTransport(IOutboxSendTransportContext context)
+        {
+            _context = context;
+        }
+
+        public ConnectHandle ConnectSendObserver(ISendObserver observer) => _context.ConnectSendObserver(observer);
+
+        public async Task Send<T>(T message, IPipe<SendContext<T>> pipe, CancellationToken cancellationToken) where T : class
+        {
+            LogContext.SetCurrentIfNull(_context.LogContext);
+
+            // check the context destination address here, versus the destination passed in with _context
+            var context = new MessageSendContext<T>(message, cancellationToken);
+
+            await pipe.Send(context).ConfigureAwait(false);
+
+            if(!context.TryGetPayload(out IOutboxTransportRepository repository))
+            {
+                throw new OutboxException("Couldn't resolve outbox repository. Configuration is incorrect.");
+            }
+
+            if(context.RequestId.GetValueOrDefault() != default)
```

Comment:
If it's a RequestClient or contains the specific header "MT-Outbox-Sweeper", than short circuit so the real transport is called. I'm open to suggestions if there's a better way.

Reply:
I've thought about this, and was thinking like the OutboxContext, we could have the request client check for it, and call through it instead of the outbox, or have the request client add a payload type that the outbox could check for and bypass if present.

Because a saga may set a RequestId/ResponseAddress and not be the request client.